### PR TITLE
✨🐛🚨 Source Google Ads: add new streams + convert list of string to list of dict

### PR DESCRIPTION
**Pull request by KimPlv**
## What
**Add new streams**
-  bidding_strategy
- lead_form_submission_data
- ad_group_ad_performance_report
- ad_group_audience_performance_report
- ad_group_performance_report
- campaign_performance_report
- geographic_performance_report
- keywords_performance_report

**Add new fields in streams**
- account_performance_report
- ad_group
- ad_group_ad
- ad_group_ad_legacy
- campaign

🚨 **Add new function to convert list of string to list of dict**
function : `convert_array` in `airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py`
example : 
- previous version : 
` "campaign.url_custom_parameters": ["key: \"medium\"\nvalue: \"cpc\"\n", "key: \"source\"\nvalue: \"adwords\"\n"]`    
- new version : 
`"campaign.url_custom_parameters": [{"key": "medium", "value": "cpc"}, {"key": "source", "value": "adwords"}]`

## 🚨 Breacking Change 
Impact on the schema of streams whose have list of string as key value as a field 
not impacted on simple list of string, or other field type

## Recommended reading order
1. `y.python`

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- Connector version is set to `3.3.2`
    - `Dockerfile` has version `3.3.2`
- Documentation updated
    - `docs/integrations/source/google-ads.md` including changelog with an entry for the initial version. 
    - `docs/integrations/README.md`

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello @KimPlv thanks for the contribution. I requested to our connector team to review your it.

File: airbyte-integrations/connectors/source-google-ads/metadata.yaml
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-ads/metadata.yaml:65]: why those streams should be in suggested list?

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-google-ads/metadata.yaml
Comment (KimPlv) [reply airbyte-integrations/connectors/source-google-ads/metadata.yaml]: changed

File: airbyte-integrations/connectors/source-google-ads/Dockerfile
Comment (KimPlv) [reply airbyte-integrations/connectors/source-google-ads/Dockerfile]: deleted

**Comment by xhensiladoda** [comment]
Hello everyone! May I ask what are the plans to merge this pull request? In my team we are exactly missing what is in this pull request. Are there any plans to unblock it ☺️ Can we help somehow to move further with it?

**Review by fidel-perez — APPROVED** [review]

**Review by midavadim — CHANGES_REQUESTED** [review]
I am tending to ask you to revert changes related to "string convertion to dict".
Reasons:

1. Current code does not properly convert strings with nested structutures like:
	ad_group_ad_legacy
	
	"ad_group_ad.ad.responsive_search_ad.descriptions": [ "text: \"Behind The Scenes: Testing The Airbyte Maintainer Program\"\nasset_performance_label: PENDING\npolicy_summary_info {\n  review_status: REVIEWED\n  approval_status: APPROVED\n}\n"]
  
    so output should be:
	
	"ad_group_ad.ad.responsive_search_ad.descriptions": [
		{
			text: Consolidate your data in your data warehouses, lakes and databases
			asset_performance_label: PENDING
			policy_summary_info {
				review_status: REVIEWED
				approval_status: APPROVED
			}
		}
	]

2. Schema changes: some type of fields was chnaged from "list of strings" to "list of objects" without specifying fields for "objects". It will lead to empty values on destination side. Instead, it is better to have a string with full content
	
	IN MASTER:
    "ad_group_ad.ad.responsive_search_ad.descriptions": {
      "type": [
        "null",
        "array"
      ],
      "items": {
        "type": "string"
      }
    },
	
	IN PR:
    "ad_group_ad.ad.responsive_search_ad.descriptions": {
      "type": [
        "null",
        "array"
      ],
      "items": {
        "type": "object"
      }
    },
	

So we should better revert changed related to "string converion" in python code and in schema.

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py
Comment (midavadim) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py:217]: I am tending to ask you to revert changes related to "string convertion to dict".
Reasons:

1. Current code does not properly convert strings with nested structutures like:
	ad_group_ad_legacy
	
	"ad_group_ad.ad.responsive_search_ad.descriptions": [ "text: \"Behind The Scenes: Testing The Airbyte Maintainer Program\"\nasset_performance_label: PENDING\npolicy_summary_info {\n  review_status: REVIEWED\n  approval_status: APPROVED\n}\n"]
  
    so output should be:
	
	"ad_group_ad.ad.responsive_search_ad.descriptions": [
		{
			text: Consolidate your data in your data warehouses, lakes and databases
			asset_performance_label: PENDING
			policy_summary_info {
				review_status: REVIEWED
				approval_status: APPROVED
			}
		}
	]

2. Schema changes: some type of fields was chnaged from "list of strings" to "list of objects" without specifying fields for "objects". It will lead to empty values on destination side. Instead, it is better to have a string with full content
	
	IN MASTER:
    "ad_group_ad.ad.responsive_search_ad.descriptions": {
      "type": [
        "null",
        "array"
      ],
      "items": {
        "type": "string"
      }
    },
	
	IN PR:
    "ad_group_ad.ad.responsive_search_ad.descriptions": {
      "type": [
        "null",
        "array"
      ],
      "items": {
        "type": "object"
      }
    },
	

So we should better revert changed related to "string converion" in python code and in schema.

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/ad_group.json
Comment (midavadim) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/ad_group.json:6]: Could you please revert to old formatting? 
We are using it everywhere in the code.

**Review by midavadim — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
Closing as not active @KimPlv please let me know if want to continue this contribution so I can reopen it.

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py
Comment (KimPlv) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py:217]: Hello @midavadim i update my function to include your recommandation 1. you're right !
For the second point, I know it's a breaking change, but I put object because it depends on the account and the field we have. If there is no data, it just put **"ad_group_ad.ad.responsive_search_ad.descriptions": []**

For us, it was really a big issue to extract the data, and this way of doing things helped us a lot. 

Need to reopen the PR to see the changes

**Comment by KimPlv** [comment]
> Closing as not active @KimPlv please let me know if want to continue this contribution so I can reopen it.

Hello @marcosmarxm, sorry for the delay, can you reopen it so @midavadim can see the changes I have added to my branch please

**Comment by marcosmarxm** [comment]
@KimPlv it looks I can't reopen because it was forced or another branch was created.

**Comment by sheinbergon** [comment]
@marcosmarxm I need to revive this contribution, re-opening

**Comment by marcosmarxm** [comment]
The team is migrating Google Ads to low-code I'd recommend to you wait until the migration is done.

**Comment by sheinbergon** [comment]
> The team is migrating Google Ads to low-code I'd recommend to you wait until the migration is done.

Is there any ETA for that?

**Pull request closed without merging**
